bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/BusinessPartnerSearchService.kt

```diff
@@ -199,4 +215,179 @@ class BusinessPartnerSearchService(
 
         return sitePage.toDto(sitePage.content.map { it })
     }
+
+    /**
+     * @see searchBusinessPartner
+     *
+     */
+    override fun searchBusinessPartner(
+        searchRequest: LegalEntityPropertiesSearchRequest,
+        searchResultFilter: Set<BusinessPartnerSearchFilterType>?,
+        paginationRequest: PaginationRequest
+    ): PageDto<BusinessPartnerSearchResultDto> {
+
+        val includeLegalEntities = shouldInclude(searchResultFilter, BusinessPartnerSearchFilterType.IncludeLegalEntities)
+        val includeSites = shouldInclude(searchResultFilter, BusinessPartnerSearchFilterType.IncludeSites)
+        val includeAddresses = shouldInclude(searchResultFilter, BusinessPartnerSearchFilterType.IncludeAdditionalAddresses)
+
+        val results = mutableListOf<BusinessPartnerSearchResultDto>()
+
+        val matchedSite = logisticAddressRepository.searchBusinessPartner(
+            searchRequest,
+            includeLegalEntities,
+            includeSites,
+            includeAddresses
+        )
+
+        results.addAll(matchedSite.map{ searchAddressResultMapping(it) })
+        return paginateResults(results, paginationRequest)
+    }
+
+    private fun shouldInclude(
+        searchResultFilter: Set<BusinessPartnerSearchFilterType>?,
+        filterType: BusinessPartnerSearchFilterType
+    ): Boolean {
+        return searchResultFilter.isNullOrEmpty() || searchResultFilter.contains(filterType)
+    }
+
+    private fun paginateResults(
+        results: List<BusinessPartnerSearchResultDto>,
+        paginationRequest: PaginationRequest
+    ): PageDto<BusinessPartnerSearchResultDto> {
+
+        val page = paginationRequest.page
+        val size = paginationRequest.size
+
+        val start = page * size
+        val end = minOf(start + size, results.size)
+        val pagedContent = if (start < results.size) results.subList(start, end) else emptyList()
+
+        val totalElements = results.size.toLong()
+        val totalPages = if (results.isEmpty()) { 0 }
+        else {
+            (results.size / size) + if (results.size % size == 0) 0 else 1
+        }
+
+        return PageDto(
+            totalElements = totalElements,
+            totalPages = totalPages,
+            page = page,
+            contentSize = size,
+            content = pagedContent
+        )
+    }
```

Comment:
Something wrong about the pagination logic. If I have no data in my result I get a contentSize of 10 due to the pagination request size attribute.